be/src/runtime/stream_load/stream_load_pipe.h

```diff
@@ -119,18 +122,23 @@ class StreamLoadPipe : public MessageBodySink {
         while (bytes_read < *data_size) {
             if (_read_buf == nullptr || !_read_buf->has_remaining()) {
                 std::unique_lock<std::mutex> l(_lock);
-                _get_cond.wait(l, [&]() { return _cancelled || _finished || !_buf_queue.empty(); });
+                _get_cond.wait_for(l, std::chrono::seconds(3),
```

Comment:
Why change the `wait` to `wait_for` ?

Reply:
Because if the stream_load stuck here for very long time(when there exists some problem for network), then the file data source `get_next` will be blocked, and this will eventually leads to the blocking of the file read threads.

If more than 48 file read threads(here suppose we still use the pipeline_hdfs_scan_thread_pool_thread_num) blocks here, then the whole thread pool will be blocked due to io.